### ISSUE TITLE
Syntax Error Near ''

### ISSUE THREAD

**Issue by BinaryAura** [HeidiSQL/HeidiSQL#827]
## Steps to reproduce this issue

In a Query Tab:

```SQL
CREATE FUNCTION CountOccurrences
(
   search LONGTEXT,
   term varchar(1024)
)
RETURNS LONGTEXT
BEGIN
  return REPLACE(search,term,'a');
END
```

## Current behavior

```
SQL Error (1064): You have an error in your SQL syntax;
check the manual that corresponds to your MySQL
server version for the right syntax to use near '' at line 8
```

## Expected behavior

Successful Creation of function CountOccurrences in mysql.func Table

## Possible solution

All solutions for this error that I've found are related to CR LF

## Environment

Windows 10 Pro (10.0.18362)
x64
Laragon Mint 4.0.14 190508
MySQL: 5.1..72
HiediSQL Portable 10.3.0.5782

**Comment by BinaryAura** [comment]
This is successful on http://sqlfiddle.com/#!9/9eecb (MySQL 5.6)

**Comment by ansgarbecker** [comment]
You need to change the query delimiter from semicolon to some other. Click the button on the main toolbar with a semicolon on it to get a prompt for it.

**Comment by BinaryAura** [comment]
I got this to work by:

changing the query to;

```SQL
CREATE FUNCTION binaryaura.CountOccurrances (src LONGTEXT, term VARCHAR(1024)) RETURNS INT
  return (LENGTH(src)-LENGTH(REPLACE(src, term, '')))/LENGTH(term)
;
```

where binaryaura is the database name.

I attempted to drop the function to test your solution and I now get

```
SQL Error (1304): FUNCTION CountOccurences already
exists
```

when I attempt to run:

```SQL
CREATE FUNCTION CountOccurrences
(
   search LONGTEXT,
   term varchar(1024)
)
RETURNS LONGTEXT
BEGIN
  return REPLACE(search,term,'a');
END
```

and

```
SQL Error (1305): FUNCTION
binaryaura.CountOccurrances does not exist
```

for

```SQL
DROP FUNCTION CountOccurances
```

Which one is it MySQL.

**Comment by ansgarbecker** [comment]
So you did not find or click the "Delimiter" button? Here you can see where it is:

![grafik](https://user-images.githubusercontent.com/7986591/71186463-8b353d80-227d-11ea-8adf-5f0a644bb779.png)

**Comment by BinaryAura** [comment]
I did and that worked but before that I started getting Heidi errors clicking on just about everything, so I restarted. The function was still in the schema so I dropped it again and then it worked. I tried using the Stored Routine form and I got errors there... So I went back to the previous method and used the form to alter. That seemed to work. I don't think I like SQL... Thanks for being so understanding. Some of this didn't seem to fit here but, SQL docs are super confusing and stack overflow are pompous jerks.
